docs/content/latest/api/ysql/datatypes/type_array/literals/text-typecasting-and-literals.md

```diff
@@ -0,0 +1,242 @@
+
+We see already, then, that the rules for composing a `numeric` literal and a `text` literal are different:
+
+- You compose a `numeric` literal by following the bare text that specifies the intended value with the `::numeric` typecast operator.
+
+- You compose a `text` literal by enquoting the bare text that specifies the intended value (however you choose to do the quoting) and by then following this with the `::text` typecast operator.
+
+(If you did, foolishly, enquote the bare text in a `numeric` literal then you would _not_ see an error. Rather, you would get implicit but undesirable behavior: first a genuine `text` value would be generated internally and then this, in turn, would be typecasted to the `numeric` value.)
+
+We've already seen, informally, some examples of array literals. Here is the rule:
+
+- You compose the bare text that specifies the intended value by writing an utterance in a dedicated grammar that starts with the left curly brace and ends with the right curly brace. (This grammar is the focus of the remaining sections in the present enclosing section.) Then you enquote this bare text (however you choose to do the quoting) and then you typecast it to the desired target array data type.
+
+These are three special cases of a more general rule. In some cases (for example in the literal for a _"row"_ type value) the enquoting mechanism might be optional (depending on the intended value) and, when written uses _double quote_ as the the enquoting character. But here, too, the general rule is the same. The bare text that specifies the intended value can always be correctly written as the `::text` typecast of that value.
+
+## Stating the general rule
+
+Here is the general rule.
+
+- The literal for a value of any data type is the possibly enquoted bare text that specifies the intended value followed by the typecast operator to the desired target data type.
+- This rule is applied recursively, for the literal for a composite value, but with different actual rules at different levels of nesting. For example, the literal for an array value as a whole must be typecasted. But, because the data type of every value in the array is already determined, the bare text that specifies these values is _not_ typecasted.
+- The `::text` typecast of any value can always be used as the bare text of the literal that will recreate that value.
+
+A simple way to see examples of the bare text that specifies an array value, when used as a component in the literal, is to create the value using the constructor and then to inspect its `::text` typecast. But, of course, the safe way to create the text of a literal for an intended value is to understand the syntax and semantics that govern its composition.
+
+When we need to make this distinction, throughout the whole of this major _"[Arrays](../../../type_array/)"_ section, we shall distinguish between:
```

Comment:
```suggestion
When this difference is useful, the [Arrays](../../../type_array/) section distinguishes between:
```

Reply:
"When this difference is useful, the..." No. "When this difference is important, the..."

I'll change this later on my laptop.